resolver/resolver.cc

```diff
@@ -1553,6 +1553,24 @@ class ResolveSignaturesWalk {
     }
 
     unique_ptr<ast::ClassDef> preTransformClassDef(core::Context ctx, unique_ptr<ast::ClassDef> original) {
+        // At this point, the type members have been fully processed, so the
+        // AttachedClass template can be resolved
+        auto klass = original->symbol;
+
+        // The resolve constants pass ensures that the singleton exists already
+        auto singleton = klass.data(ctx)->lookupSingletonClass(ctx);
+        ENFORCE(singleton.exists());
+
+        // When AttachedClass is present on the singleton, update its upper
+        // bound now that all of the other type members have been defined.
+        auto attachedClass = singleton.data(ctx)->findMember(ctx, core::Names::Constants::AttachedClass());
+        if (attachedClass.exists()) {
```

Comment:
Why wouldn't it exist?

Reply:
I think this is an artifact from when I was moving this around. It will always exist if the singleton does as well, so I'll remove the `if`.